javalib/src/main/scala/java/math/BigInteger.scala

```diff
@@ -0,0 +1,1500 @@
+    }
+    val divisorSign = divisor.sign
+    if (divisor.isOne) {
+      if (divisor.sign > 0) this else this.negate()
+    } else {
+      val thisSign = sign
+      val thisLen = numberLength
+      val divisorLen = divisor.numberLength
+      if (thisLen + divisorLen == 2) {
+        var bi = (digits(0) & 0xFFFFFFFFL) / (divisor.digits(0) & 0xFFFFFFFFL)
+        if (thisSign != divisorSign) {
+          bi = -bi
+        }
+        valueOf(bi)
+      } else {
+        val cmp = if (thisLen != divisorLen) if (thisLen > divisorLen) 1 else -1
+                  else Elementary.compareArrays(digits, divisor.digits, thisLen)
+
+        if (cmp == EQUALS) {
+          if (thisSign == divisorSign) ONE else MINUS_ONE
+        } else if (cmp == LESS) {
+          ZERO
+        } else {
+          val resLength = thisLen - divisorLen + 1
+          val resDigits = Array.ofDim[Int](resLength)
+          val resSign = (if ((thisSign == divisorSign)) 1 else -1)
+          if (divisorLen == 1) {
+            Division.divideArrayByInt(resDigits, digits, thisLen, divisor.digits(0))
+          } else {
+            Division.divide(resDigits, resLength, digits, thisLen, divisor.digits, divisorLen)
+          }
+          val result = new BigInteger(resSign, resLength, resDigits)
+          result.cutOffLeadingZeroes()
+          result
+        }
+      }
+    }
+  }
+
+  /**
+   * Returns a {@code BigInteger} array which contains {@code this / divisor} at
+   * index 0 and {@code this % divisor} at index 1.
+   *
+   * @param divisor value by which {@code this} is divided.
+   * @return {@code [this / divisor, this % divisor]}.
+   * @throws NullPointerException if {@code divisor == null}.
+   * @throws ArithmeticException if {@code divisor == 0}.
+   * @see #divide
+   * @see #remainder
+   */
+  def divideAndRemainder(divisor: BigInteger): Array[BigInteger] = {
+    val divisorSign = divisor.sign
+    if (divisorSign == 0) {
+      throw new ArithmeticException("BigInteger divide by zero")
+    }
+    val divisorLen = divisor.numberLength
+    val divisorDigits = divisor.digits
+    if (divisorLen == 1) {
+      Division.divideAndRemainderByInteger(this, divisorDigits(0), divisorSign)
+    } else {
+      // res[0] is a quotient and res[1] is a remainder:
+      val thisDigits = digits
+      val thisLen = numberLength
+      val cmp = if (thisLen != divisorLen) if (thisLen > divisorLen) 1 else -1
+                else Elementary.compareArrays(thisDigits, divisorDigits, thisLen)
+
+      if (cmp < 0) {
+        Array(ZERO, this)
+      } else {
+        val thisSign = sign
+        val quotientLength = thisLen - divisorLen + 1
+        val remainderLength = divisorLen
+        val quotientSign = (if ((thisSign == divisorSign)) 1 else -1)
+        val quotientDigits = Array.ofDim[Int](quotientLength)
+        val remainderDigits = Division.divide(quotientDigits, quotientLength, thisDigits, thisLen, divisorDigits,
+          divisorLen)
```

Comment:
Cut this line earlier so that you stay under 80 characters. Indent 4 spaces on the continuation line.